packages/block-library/src/heading/edit.js

```diff
@@ -28,14 +30,37 @@ function HeadingEdit( {
 	mergedStyle,
 	clientId,
 } ) {
-	const { textAlign, content, level, placeholder } = attributes;
+	const { textAlign, content, level, placeholder, anchor } = attributes;
 	const tagName = 'h' + level;
 	const blockProps = useBlockProps( {
 		className: classnames( {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
 		} ),
 		style: mergedStyle,
 	} );
+	const allHeadingAnchors = useAllHeadingAnchors( clientId );
```

Comment:
This is where I worry the most about performance as it runs too many times I think.. Any thoughts @mcsf , @ellatrix ?

Reply:
The implementation was refactored just a few minutes ago in commit https://github.com/WordPress/gutenberg/pull/30825/commits/5bb9302bb8bc3f0f87ad793c25bfea38fe2e6ae2, I believe this is no longer an issue.
The `useAllHeadingAnchors` and all the content-parsing no longer occurs. :+1: